src/v/model/async_adl_serde.cc

```diff
@@ -64,7 +66,9 @@ async_adl<model::record_batch>::from(iobuf_parser& in) {
     auto recs = std::vector<model::record>{};
     recs.reserve(hdr.bhdr.record_count);
     for (int i = 0; i < hdr.bhdr.record_count; ++i) {
-        recs.push_back(co_await async_adl<model::record>{}.from(in));
+        auto rec = adl<model::record>{}.from(in);
+        recs.push_back(std::move(rec));
+        co_await ss::coroutine::maybe_yield();
```

Comment:
interesting. from a scheduling POV this looks identical to before. is there some overhead in the async_adl version of recording encoding that we're avoiding here?

Reply:
It's been a while since I originally made this change.  I think that maybe the original co_await/async_adl call ended up not being a scheduling point at all because it was just executing synchronously and returning a ready future.  That's me retroactively making up a reason though.

Reply:
seems like that would require the compiler to know that a coroutine always returned an immediately ready future. anyway, this version looks good just curious about it.